src/Rings/preliminaries.jl

```diff
@@ -0,0 +1,18 @@
+########################################################################
+# This file contains declarations of functions and types which need 
+# to come before the later inclusions in Rings.jl
+########################################################################
+
+@doc raw"""
+    is_local(R::Ring)
+
+Return whether a given ring `R` is a local ring.
+"""
+function is_local(R::Ring)
+  error("check whether $R is local is not implemented")
+end
+
+is_local(::Field) = true
```

Comment:
This looks like something that really should be in AbstractAlgebra?

Reply:
Of course, you could take the two dummy-functions to AbstractAlgebra, leave the meaningful functions for localized polynomial rings here (because they need the localization framework) and maybe give the right answer for (true) for power series rings in AbstractAlgebra again. But is the question of locality even relevant in the scope of Abstract Algebra? Doesn't it make more sense to add the short dummy functions right where the question arises in a meaningful context?

Reply:
It probably should. However, in the interest of being able to progress with the overhaul of free resolutions, I suggest to move this issue to another pr.